### ISSUE TITLE
Error with opening shapefile

### ISSUE THREAD

**Issue by samantaraymadhu6** [r-spatial/sf#1501]
I am trying to execute below code, but getting the error "Error: Cannot open "C:\Users\Username\AppData\Local\Temp\RtmpYFM8Lo"; The source could be corrupt or not supported. See `st_drivers()` for a list of supported formats."
```r
library(tidyverse)
library(tidycensus)
library(tigris)
library(sf)
library(sp)
kenton_pop_block_2010 <- get_decennial(
  year = 2010,
  geography = "block",
  variables = "P001001",
  state = "KY",
  county = "Kenton County",
   geometry = TRUE
) %>%
  select(GEOID, value, year) %>%
  rename(population = value)%>%
  mutate(year = 2010)
```
Appreciate your help!

**Comment by edzer** [comment]
The call to `get_decennial` succeeded for me, but `select` failed, with the error that `year` is not present.

**Comment by samantaraymadhu6** [comment]
Okay I got the same error that year is not present when I tried with geometry = FALSE.But I want the code to be executed with geometry = TRUE

**Comment by edzer** [comment]
I see:
```r
> library(tidyverse)
── Attaching packages ─────────────────────────────────────── tidyverse 1.3.0 ──
✔ ggplot2 3.3.2     ✔ purrr   0.3.4
✔ tibble  3.0.3     ✔ dplyr   1.0.2
✔ tidyr   1.1.2     ✔ stringr 1.4.0
✔ readr   1.3.1     ✔ forcats 0.5.0
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
> library(tidycensus)
> library(tigris)
To enable 
caching of data, set `options(tigris_use_cache = TRUE)` in your R script or .Rprofile.
> library(sf)
Linking to GEOS 3.8.0, GDAL 3.1.3, PROJ 7.1.1
> library(sp)
> kenton_pop_block_2010 <- get_decennial(
+   year = 2010,
+   geography = "block",
+   variables = "P001001",
+   state = "KY",
+   county = "Kenton County",
+    geometry = TRUE
+ )
Getting data from the 2010 decennial Census
Downloading feature geometry from the Census website.  To cache shapefiles for use in future sessions, set `options(tigris_use_cache = TRUE)`.
Using FIPS code '21' for state 'KY'
Using FIPS code '117' for 'Kenton County'
  |======================================================================| 100%
> kenton_pop_block_2010
Simple feature collection with 3778 features and 4 fields
geometry type:  MULTIPOLYGON
dimension:      XY
bbox:           xmin: -84.62375 ymin: 38.79009 xmax: -84.41744 ymax: 39.09998
geographic CRS: NAD83
# A tibble: 3,778 x 5
   GEOID   NAME               variable value                            geometry
   <chr>   <chr>              <chr>    <dbl>                  <MULTIPOLYGON [°]>
 1 211170… Block 4013, Block… P001001      0 (((-84.58809 39.05257, -84.58839 3…
 2 211170… Block 2012, Block… P001001      0 (((-84.52916 39.01465, -84.52918 3…
 3 211170… Block 1012, Block… P001001     14 (((-84.49077 39.03645, -84.49077 3…
 4 211170… Block 1007, Block… P001001      1 (((-84.51315 39.08069, -84.51301 3…
 5 211170… Block 1007, Block… P001001      0 (((-84.52129 39.08759, -84.52128 3…
 6 211170… Block 1007, Block… P001001    224 (((-84.55898 39.06576, -84.55925 3…
 7 211170… Block 2029, Block… P001001     31 (((-84.57516 39.06263, -84.57514 3…
 8 211170… Block 1007, Block… P001001    326 (((-84.60001 39.01088, -84.59979 3…
 9 211170… Block 2042, Block… P001001      0 (((-84.53464 39.02567, -84.5344 39…
10 211170… Block 3003, Block… P001001      1 (((-84.49294 39.00909, -84.49292 3…
# … with 3,768 more rows
```

**Comment by samantaraymadhu6** [comment]
Thanks, I tried now with all the packages installed, got the below error. Any idea what is the issue with sf package?
 kenton_pop_block_2010 <- get_decennial(
+   year = 2010,
+   geography = "block",
+   state = "KY",
+   county = "Kenton County",
+   variables = "P001001",
+   geometry = TRUE
+   )
Getting data from the 2010 decennial Census
Downloading feature geometry from the Census website.  To cache shapefiles for use in future sessions, set `options(tigris_use_cache = TRUE)`.
Using FIPS code '21' for state 'KY'
Using FIPS code '117' for 'Kenton County'
Error: Cannot open "/private/var/folders/gp/c4k02s8n4cg25j2_w4xbc0n00000gn/T/RtmpFw9jcV"; The source could be corrupt or not supported. See `st_drivers()` for a list of supported formats.
In addition: Warning message:
In unzip(file_loc, exdir = tmp) : error 1 in extracting from zip file
>

**Comment by edzer** [comment]
No. Maybe time to bring this up with the `tigris` author?

**Comment by samantaraymadhu6** [comment]
Okay so what I did is I uninstalled sf and installed again. The open error is gone ,but I am getting the below error. I also changed geography = "county" and the whole chunk works fine. 
 
`> kenton_pop_block_2010 <- get_decennial(
+   year = 2010,
+   geography = "block",
+   state = "KY",
+   county = "Kenton County",
+   variables = "P001001",geometry = TRUE
+   )
Getting data from the 2010 decennial Census
Downloading feature geometry from the Census website.  To cache shapefiles for use in future sessions, set `options(tigris_use_cache = TRUE)`.
Using FIPS code '21' for state 'KY'
Using FIPS code '117' for 'Kenton County'
Cannot open layer tl_2010_21117_tabblock10
Error in CPL_read_ogr(dsn, layer, query, as.character(options), quiet,  : 
  Opening layer failed`
